config/routes.rb

```diff
@@ -26,6 +26,8 @@
     post 'auth/callback'
     get  'auth/sign_out'
 
+    post '/vote', to: 'votes#create', as: 'vote'
```

Comment:
Since this should only ever be called via an XHR client-side request (because you have `remote: true` on the link), it may be worth locking down by adding the following to this route:

```ruby
constraints: { format: :js }
```

… then in the controller you _should_ be able to do something like:

```ruby
respond_to do |format|
  format.js { redirect_to course_path(vote.course_slug) }
end
```

… Turbolinks will automatically pick this up and handle the redirect client side.

This way you don't need that empty view, and it's locked down accordingly.

Reply:
Done.